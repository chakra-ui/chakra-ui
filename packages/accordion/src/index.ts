export * from "./accordion"
export * from "./use-accordion"
