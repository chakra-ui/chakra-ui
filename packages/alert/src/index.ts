export { Alert, type AlertProps } from "./alert"
export { useAlertStyles, type AlertStatus } from "./alert-context"
export {
  AlertDescription,
  type AlertDescriptionProps,
} from "./alert-description"
export { AlertIcon, type AlertIconProps } from "./alert-icon"
export { AlertTitle, type AlertTitleProps } from "./alert-title"
