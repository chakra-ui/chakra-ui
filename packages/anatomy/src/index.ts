import { anatomy } from "@chakra-ui/theme-tools"

/**
 * **Accordion anatomy**
 * - Root: the root container of the accordion
 * - Container: the accordion item contains the button and panel
 * - Button: the button is the trigger for the panel
 * - Panel: the panel is the content of the accordion item
 * - Icon: the expanded/collapsed icon
 */
export const accordionAnatomy = anatomy("accordion")
  .parts("root", "container", "button", "panel")
  .extend("icon")

/**
 * **Alert anatomy**
 * - Title: the alert's title
 * - Description: the alert's description
 * - Icon: the alert's icon
 */
export const alertAnatomy = anatomy("alert")
  .parts("title", "description", "container")
  .extend("icon")

/**
 * **Avatar anatomy**
 * - Container: the container for the avatar
 * - Label: the avatar initials text
 * - Excess Label: the label or text that represents excess avatar count.
 * Typically used in avatar groups.
 * - Group: the container for the avatar group
 */
export const avatarAnatomy = anatomy("avatar")
  .parts("label", "badge", "container")
  .extend("excessLabel", "group")

/**
 * **Breadcrumb anatomy**
 * - Item: the container for a breadcrumb item
 * - Link: the element that represents the breadcrumb link
 * - Container: the container for the breadcrumb items
 * - Separator: the separator between breadcrumb items
 */
export const breadcrumbAnatomy = anatomy("breadcrumb")
  .parts("link", "item", "container")
  .extend("separator")

export const buttonAnatomy = anatomy("button").parts()

export const checkboxAnatomy = anatomy("checkbox")
  .parts("control", "icon", "container")
  .extend("label")

export const circularProgressAnatomy = anatomy("progress")
  .parts("track", "filledTrack")
  .extend("label")

export const drawerAnatomy = anatomy("drawer")
  .parts("overlay", "dialogContainer", "dialog")
  .extend("header", "closeButton", "body", "footer")

export const editableAnatomy = anatomy("editable").parts(
  "preview",
  "input",
  "textarea",
)

export const formAnatomy = anatomy("form").parts(
  "container",
  "requiredIndicator",
  "helperText",
)

export const formErrorAnatomy = anatomy("formError").parts("text", "icon")

export const inputAnatomy = anatomy("input").parts("addon", "field", "element")

export const listAnatomy = anatomy("list").parts("container", "item", "icon")

export const menuAnatomy = anatomy("menu")
  .parts("button", "list", "item")
  .extend("groupTitle", "command", "divider")

export const modalAnatomy = anatomy("modal")
  .parts("overlay", "dialogContainer", "dialog")
  .extend("header", "closeButton", "body", "footer")

export const numberInputAnatomy = anatomy("numberinput").parts(
  "root",
  "field",
  "stepperGroup",
  "stepper",
)

export const pinInputAnatomy = anatomy("pininput").parts("field")

export const popoverAnatomy = anatomy("popover")
  .parts("content", "header", "body", "footer")
  .extend("popper", "arrow", "closeButton")

export const progressAnatomy = anatomy("progress").parts(
  "label",
  "filledTrack",
  "track",
)

export const radioAnatomy = anatomy("radio").parts(
  "container",
  "control",
  "label",
)

export const selectAnatomy = anatomy("select").parts("field", "icon")

export const sliderAnatomy = anatomy("slider").parts(
  "container",
  "track",
  "thumb",
  "filledTrack",
)

export const statAnatomy = anatomy("stat").parts(
  "container",
  "label",
  "helpText",
  "number",
  "icon",
)

export const switchAnatomy = anatomy("switch").parts(
  "container",
  "track",
  "thumb",
)

export const tableAnatomy = anatomy("table").parts(
  "table",
  "thead",
  "tbody",
  "tr",
  "th",
  "td",
  "tfoot",
  "caption",
)

export const tabsAnatomy = anatomy("tabs").parts(
  "root",
  "tab",
  "tablist",
  "tabpanel",
  "tabpanels",
  "indicator",
)

/**
 * **Tag anatomy**
 * - Container: the container for the tag
 * - Label: the text content of the tag
 * - closeButton: the close button for the tag
 */
export const tagAnatomy = anatomy("tag").parts(
  "container",
  "label",
  "closeButton",
)
