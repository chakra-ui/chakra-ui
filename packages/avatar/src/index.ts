export * from "./avatar"
export * from "./avatar-group"
