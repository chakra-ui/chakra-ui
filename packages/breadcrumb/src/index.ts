export { Breadcrumb, type BreadcrumbProps } from "./breadcrumb"
export { BreadcrumbItem, type BreadcrumbItemProps } from "./breadcrumb-item"
export {
  BreadcrumbSeparator,
  type BreadcrumbSeparatorProps,
} from "./breadcrumb-separator"
export { BreadcrumbLink, type BreadcrumbLinkProps } from "./breadcrumb-link"
export { useBreadcrumbStyles } from "./breadcrumb-context"
export type { BreadcrumbOptions } from "./breadcrumb-types"
