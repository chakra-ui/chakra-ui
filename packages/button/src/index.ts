export * from "./button"
export * from "./button-group"
export * from "./icon-button"
export * from "./button-spinner"
