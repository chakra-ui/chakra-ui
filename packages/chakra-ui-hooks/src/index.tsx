export * from "./useAccordion/useAccordion";
export { default as useAriaHidden } from "./useModal/utils/useAriaHidden";
export { default as useBlurOutside } from "./useBlurOutside";

export { default as useCheckbox } from "./useCheckbox/useCheckbox";
export * from "./useCheckboxGroup/useCheckboxGroup";
export {
  default as useCheckboxGroup,
} from "./useCheckboxGroup/useCheckboxGroup";
export { default as useCounter } from "./useCounter/useCounter";
export { default as useClipboard } from "./useClipboard/useClipboard";
export { default as useControllableValue } from "./useControllableValue";
export { default as createCtx } from "./useCreateContext";

export { default as useDisclosure } from "./useDisclosure/useDisclosure";

export { default as useIsomorphicEffect } from "./useIsomorphicEffect";

export { default as useFocusEffect } from "./useFocusEffect/useFocusEffect";
export { default as useFocusOnHide } from "./useFocusOnHide";
export { default as useFocusOnShow } from "./useFocusOnShow";
export * from "./useForkRef";

export { default as useId } from "./useId";
export { default as useIds } from "./useIds";

export {
  default as useLockBodyScroll,
} from "./useLockBodyScroll/useLockBodyScroll";
export { default as useLogger } from "./useLogger";

// export * from "./useNumberInput/useNumberInput";
// export { default as useNumberInput } from "./useNumberInput/useNumberInput";

export * from "./usePopper/usePopper";
export { default as usePopper } from "./usePopper/usePopper";
export { default as usePrevious } from "./usePrevious";

export * from "./useRadioGroup/useRadioGroup";
export { default as useRadioGroup } from "./useRadioGroup/useRadioGroup";
export { default as useRapidKeyDown } from "./useRapidKeydown/useRapidKeyDown";

export * from "./useSelection";
export * from "./useSlider/useSlider";

export * from "./useTabbable";
export * from "./useTabs";
export { default as useTabbable } from "./useTabbable";

export { default as useUpdateEffect } from "./useUpdateEffect";
