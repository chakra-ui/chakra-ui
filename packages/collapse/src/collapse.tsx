/** @jsx jsx */
import * as React from "react"
import AnimateHeight, {
  AnimateHeightProps as AnimateProps,
} from "react-animate-height"
import { chakra, jsx, PropsOf } from "@chakra-ui/system"
import { __DEV__ } from "@chakra-ui/utils"

type AnimateHeightProps = Pick<
  AnimateProps,
  | "animationStateClasses"
  | "applyInlineTransitions"
  | "delay"
  | "easing"
  | "style"
  | "children"
>

export interface CollapseOptions {
  /**
   * If `true`, the content will be visible
   */
  isOpen?: boolean
  /**
   * If `true`, the opacity of the content will be animated
   */
  animateOpacity?: boolean
  /**
   * The duration of the animation in `ms`
   */
  duration?: number
  /**
   * The height you want the content in it's collapsed state. Set to `0` by default
   */
  startingHeight?: number | string
  /**
   * The height you want the content in it's expanded state. Set to `auto` by default
   */
  endingHeight?: number | string
  /**
   * The function to be called when the collapse animation starts. It provides the `newHeight` as an argument
   */
  onAnimationEnd?(props: { newHeight: number }): void
  /**
   * The function to be called when the collapse animation ends. It provides the `newHeight` as an argument
   */
  onAnimationStart?(props: { newHeight: number }): void
}

export type ICollapse = CollapseProps

export type CollapseProps = AnimateHeightProps &
  CollapseOptions &
  PropsOf<typeof chakra.div>

export const Collapse = React.forwardRef(function Collapse(
  props: CollapseProps,
  ref: React.Ref<any>,
) {
  const {
    isOpen,
    animateOpacity = true,
    onAnimationStart,
    onAnimationEnd,
    duration,
    easing = "ease",
    startingHeight = 0,
    endingHeight = "auto",
    ...rest
  } = props

  return (
    <AnimateHeight
      duration={duration}
      easing={easing}
      animateOpacity={animateOpacity}
      height={isOpen ? endingHeight : startingHeight}
      applyInlineTransitions={false}
      onAnimationStart={onAnimationStart}
      onAnimationEnd={onAnimationEnd}
      sx={{
        transition:
          "height .2s ease,opacity .2s ease-in-out,transform .2s ease-in-out",
        "&.rah-animating--to-height-zero": {
          opacity: 0,
          transform: "translateY(-0.625rem)",
        },
      }}
    >
      <chakra.div ref={ref} {...rest} />
    </AnimateHeight>
  )
})

if (__DEV__) {
  Collapse.displayName = "Collapse"
}

export default Collapse
