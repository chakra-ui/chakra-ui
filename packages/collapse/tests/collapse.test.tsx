import * as React from "react"
import { render } from "@chakra-ui/test-utils"
import { Collapse } from "../src"

test("Collapse renders correctly", () => {
  const utils = render(<Collapse>sfsjdfkbfkjdfdjksf</Collapse>)
  expect(utils.asFragment()).toMatchSnapshot()
})
