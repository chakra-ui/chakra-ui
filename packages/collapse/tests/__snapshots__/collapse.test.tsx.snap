// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders correctly 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-transition: height .2s ease,opacity .2s ease-in-out,-webkit-transform .2s ease-in-out;
  -webkit-transition: height .2s ease,opacity .2s ease-in-out,transform .2s ease-in-out;
  transition: height .2s ease,opacity .2s ease-in-out,transform .2s ease-in-out;
}

.emotion-1.rah-animating--to-height-zero {
  opacity: 0;
  -webkit-transform: translateY(-0.625rem);
  -ms-transform: translateY(-0.625rem);
  transform: translateY(-0.625rem);
}

<div
    aria-hidden="true"
    class="rah-static rah-static--height-zero emotion-1"
    style="height: 0px; overflow: hidden;"
  >
    <div
      style="transition: opacity 250ms ease 0ms; opacity: 0; display: none;"
    >
      <div
        class="emotion-0"
      >
        sfsjdfkbfkjdfdjksf
      </div>
    </div>
  </div>
</DocumentFragment>
`;
